pkg/proxy/proxy/transparent.go

```diff
@@ -175,6 +200,14 @@ func (t *TransparentProxyService) ServeMetadata(addr string) {
 	}
 }
 
+// Check that the requested url is allowed by the network policy.
+func (proxy TransparentProxyService) CheckNetworkPolicy(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
```

Comment:
nit: since this is actually modifying the request/response, I'd probably call this "ApplyNetworkPolicy." something called "check" i expect to return a bool or an error.